### DISCUSSION TITLE
How to read large files

### DISCUSSION THREAD

**Discussion by smileyboy2019** [langchain4j/langchain4j#2030] [Ideas]
How to read large files, such as uploading a file, so that the large model can understand the entire text instead of segmenting it
Do we need to use tools to process documents. Also, does the large model support larger PDFs or Word files. Do you need to choose Rag technology or submit a large model response directly.

**Comment by smileyboy2019** [comment]
Can we use tools to directly submit files to LLM,

**Comment by glaforge** [comment]
You might want to have a look at models like Google's Gemini that can accept up to 2 million tokens in their context windows.
https://blog.google/technology/ai/google-gemini-next-generation-model-february-2024/#context-window

Gemini can ingest full books in one single prompt!

**Comment by smileyboy2019** [comment]
Does the API interface support the transmission of large amounts of data when using Olama as a local model. Prompt ":" Supports maximum word count for transmission "
